plugin/src/main/java/com/denizenscript/denizen/utilities/blocks/CuboidBlockSet.java

```diff
@@ -375,25 +422,64 @@ public void flipX() {
     public void flipY() {
         FullBlockData[] bd = new FullBlockData[blocks.length];
         int index = 0;
-        center_y = y_length - center_y;
+        center_y = y_length - center_y - 1;
         for (int x = 0; x < x_width; x++) {
             for (int y = y_length - 1; y >= 0; y--) {
                 for (int z = 0; z < z_height; z++) {
-                    bd[index++] = blockAt(x, y, z);
+                    bd[index++] = blockAt(x, y, z).flipY();
                 }
             }
         }
         blocks = bd;
     }
 
+    public void flipEntitiesZ() {

```

Comment:
These two methods look like they can be easily merged into one, as code is mostly the same other than
A: Numeric values that can be passed as params, and
B: What I assume are unintentional differences brought about while testing? I note for example `X` seems to handle `duplicate` smarter than `Z` version

Reply:
Reworked somewhat to combine X/Z versions, and pass in X/Z offset multipliers to determine flip. ab8af60022dc9f16118b9fb2f1429cf94882389f